portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -12,15 +17,42 @@ public class FormHandlerServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+
+    if(!request.getParameter("text-input-name").matches("[\\w*\\s*]*")){
+        request.setAttribute("error", "Please enter only letters, numbers, and spaces for names.");
+        response.getWriter().println("Please enter only letters, numbers, and spaces for names.");
```

Comment:
I don't think setting a response for the Post does anything.

FYI: To actually handle the return from a form submit, you will have  to use jquery and submit from js. Here is a stackoverflow answer, but you don;t need to implement it. Just wanted you understand how to handle it correctly - https://stackoverflow.com/questions/30171099/is-it-possible-to-detect-form-submit-fail-in-javascript